docs/manual/index-checker.tex

```diff
@@ -457,8 +460,65 @@ \section{Binary search indices\label{index-searchindex}}
   This is the top type, and programmers should rarely need to write it.
 \end{description}
 
+\section{IndexOf indices\label{index-indexof}}
```

Comment:
I found the manual section confusing.

What is an "IndexOf index"?  The term never appears in the manual again, so this section title will confuse users.

`@IndexOfIndexFor` should be listed in section 7.1 along with other annotations.

The manual does not explicitly describe the relationship between `@LTLengthOf and `@IndexOfIndexFor`.
I would think that `@IndexOfIndexFor` should appear in section 7.2 and 7.3 to make these relationships clear.

More seriously, the manual does not explain the distinction between `@IndexOrLow` and `@IndexOfIndexFor`.  These seem to be exactly the same except for the addition of a parameter, and this should be discussed and the names should be similar.

The mnemonic behind the name `@IndexOfIndexFor` is never explained.  It's also ungainly.  I suggest renaming it.  Maybe `@IndexOf`, though that could be confused with `@IndexFor`.  Something more like `@IndexOrLow` would be appropriate.

Does the implementation require a separate checker, or is `@IndexOfIndexFor` just syntactic sugar?
The manual description strongly suggests the latter, and that is the implementation I would recommend.  It is likely to lead to fewer special cases and more correct code in general.

Even if the implementation requires a separate checker, the manual need not make the distinction.
Describing this together with the other upper bound and lower bound annotations, as syntactic sugar (exactly as `@IndexFor` is described), is probably most helpful to users.